dask/array/core.py

```diff
@@ -2650,12 +2664,34 @@ def _enforce_dtype(*args, **kwargs):
     return result
 
 
-@wraps(chunk.broadcast_to)
-def broadcast_to(x, shape):
+def broadcast_to(x, shape, chunks=None):
+    """Broadcast an array to a new shape.
+
+    Parameters
+    ----------
+    x : array_like
+        The array to broadcast.
+    shape : tuple
+        The shape of the desired array.
+    chunks : tuple, optional
+        If provided, then the result will use these chunks instead of the same
+        chunks as the source array. Setting chunks explicitly as part of
+        broadcast_to is much more efficient than rechunking afterwards.
+        Chunks are only allowed to differ from the original shape along
+        dimensions that are new on the result or have size 1 the input array.
+
+    Returns
+    -------
+    broadcast : dask array
+
+    See Also
+    --------
+    numpy.broadcast_to
+    """
     x = asarray(x)
     shape = tuple(shape)
 
-    if x.shape == shape:
+    if x.shape == shape and (chunks is None or chunks == x.chunks):
         return x
```

Comment:
So I guess this means that if the chunks are different and the shape is the same we proceed. What about warning and doing nothing, or erroring instead?

Reply:
Adding to that thought, we might want to consider what it means if `x.chunks` mismatches with `chunks[:x.ndim]`. Should this generally be giving a warning, error, or quietly rechunking the data under the hood? Likely with an exemption for dimensions of length `1`.

Reply:
NVM I guess this is handled below. Maybe it would make sense to move that check up here so it is more visible. Not sure if that is straightforward.

Reply:
I think this exists to avoid doing all the expensive validation if it isn't necessary.